docs/spec.md

```diff
@@ -367,31 +356,26 @@ to that of the derivation (or derivations).
 
 
 
-## ChainRec
+### ChainRec
 
-#### Dependencies
-
-  1. Chain
-
-#### Methods
-
-  1. `chainRec :: ChainRec m => Type m ~> ((a → c, b → c, a) → m c, a) → m b`
+```js
+ChainRec<T> {
+  chainRec: ((a => c, b => c, a) => T<c>, a) => T<b>
+}
+```
```

Comment:
Not sure introduction of `R` type is a good idea. I feel like we'll have to complement this module signature with an elaborate explanation, which we can avoid by using a not parametric type `c`. I agree this is not quite correct, but much simpler.

> You have to return the result of `next` or `done` directly

Hm, I'm pretty sure it need to be wrapped in the type. Take a closer look at fantasy-land's signature https://github.com/fantasyland/fantasy-land#chainrec-method